### ISSUE TITLE
IOS App plugin never gets deconstructed

### ISSUE THREAD

**Issue by AlexHarker** [iPlug2/iPlug2#1077]
Various calls to dealloc never happening the IOS app, and thus the plugin itself never gets deconstructed (there is a call to delete it in a dealloc, but that never happens).

This has not been tested for the plugin.

**Comment by olilarkin** [comment]
could you be more specific?

**Comment by olilarkin** [comment]
When the app gets terminated all memory is freed by the os, dealloc etc is bypassed. If you needed the ability to unload the audiounit, then that could be added to a customised IPlugAUPlayer. Hopefully the dealloc in IPlugAudioUnit would be called then. I tried the AUv3 in AUM and dealloc is called when the plugin is unloaded.

**Comment by AlexHarker** [comment]
Yes - I'm not worried about memory allocation here - I was using the app wrapper and saving some settings on quit, which is fine on desktop, but not on IOS as an app - for now I'm doing that elsewhere (which is arguably a better pattern), but importantly if anyone is assuming that whatever is constructed will get a call to the destructor then that isn't true.

Glad it works correctly in AuM - this is probably non-urgent, but my assumption was that my settings would get saved when I quit the app, but that wasn't the case.

**Comment by olilarkin** [comment]
A common thing to do on iOS is to write to the App's [userdefaults](https://developer.apple.com/documentation/foundation/nsuserdefaults?language=objc). I am currently working on persisting stuff there

**Comment by olilarkin** [comment]
probably including the last plugin state
